return 404 status in API where applicable, prevent api view crashes on invalid input

aviaryan opened the pull request:
* Fixes #320 
Return 404 status code with appropriate message when event/speaker/session/page doesn't exist 
* Fixes #321 
API views crash due to non-existant entity - Some API views like the one to get a single speaker would crash if speaker id didn't exist. This has been fixed
* Refactored code at places

All the updates have been done in accordance with the recent discussions on API.

**Note** - This PR contains #352 

PS - codecov/patch fails for this as the whole API code is *untested* [currently](https://codecov.io/gh/fossasia/open-event-orga-server/src/91d6611e6fb659a0d1722831b1d3cd936270f27c/open_event/views/views.py). My aim today was to write tests for them but this issue needed to be worked upon first before writing tests.

aviaryan commented:
@juslee Please review. Sorry the codecov/patch tests fails. To make that pass, I will have to write tests for API code which I plan to do in the coming days.

leto commented on open_event/helpers/data_getter.py:
```diff
@@ -183,3 +183,18 @@ def get_association_by_event_and_user(event_id, user_id):
+    @staticmethod
+    def get_event(event_id):
+        """Get event by id, None if it doesn't exist"""
+        return Event.query.filter_by(id=event_id).first()
```
This is more efficiently written as: `Event.query.get(event_id)` , same goes for the two instances below

leto commented:
This is a good start. The name `event_status_code` is a bit misleading. If an event exists but not a track, you will return a 200 instead of a 404. I suggest renaming the function `status_code` and if gets passed a positive integer, return 200, if it gets passed None/0, return 404. You can then pass the count of the number of objects in the JSON to `status_code`

aviaryan commented:
@leto Thanks for your review. I am working on the improvements.

> If an event exists but not a track, you will return a 200 instead of a 404.

Sorry but I don't follow. Can you provide an example ?

Also I wanted to ask, what to return when there are no tracks in `/api/v1/event/<int:event_id>/tracks`. 200 or 404 ? Currently I am returning 200.

juslee commented:
@shivamMg refer to @aviaryan question.

aviaryan commented:
Only one doubt remains in the API part. See https://github.com/fossasia/open-event-orga-server/issues/320#issuecomment-220539242

shivamMg commented:
> Also I wanted to ask, what to return when there are no tracks in /api/v1/event/<int:event_id>/tracks (event_id exists). 200 or 404 ? Currently I am returning 200.

Returning 200 with the following would be correct.

```json
{
    "tracks": []
}
```

aviaryan commented:
@juslee @shivamMg I have updated the PR with the improvements discussed in #320 and #336 . Also I have updated the first post with more details of this pull. Please review.
